apps/documents/serializers.py

```diff
@@ -80,11 +84,22 @@ def get_presigned_s3(self, obj):
         if obj.upload_status != UploadStatus.COMPLETE:
             return super().get_presigned_s3(obj)
 
+        key = '/'.join(obj.s3_path.split('/')[3:])
```

Comment:
If you look at the document model, you should be able to get the key by calling obj.s3_key instead of deconstructing s3_path.